src/VisualStudio/Core/Def/Progression/GraphProvider.cs

```diff
@@ -162,15 +164,12 @@ public void BeginGetGraphData(IGraphContext context)
 
             var graphQueries = GetGraphQueries(context, _threadingContext, _asyncListener);
 
-            if (graphQueries.Count > 0)
-            {
-                _graphQueryManager.AddQueries(context, graphQueries);
-            }
-            else
-            {
-                // It's an unknown query type, so we're done
-                context.OnCompleted();
-            }
+            // Perform the queries asynchronously  in a fire-and-forget fashion.  This helper will be responsible
+            // for always completing the context.
+            var asyncToken = _asyncListener.BeginAsyncOperation(nameof(BeginGetGraphData));
+            _ = _graphQueryManager
+                .AddQueriesAsync(context, graphQueries)
+                .CompletesAsyncOperation(asyncToken);
```

Comment:
`context.OnCompleted();` is always called within AddQueriesAsync, so there was no need to special case things on this side of the call.